src/jobflow/schemas/job_output_schema.py

```diff
@@ -0,0 +1,62 @@
+"""A Pydantic model for Jobstore document."""
+
+from typing import Generic, List, TypeVar
+
+from monty.json import MontyDecoder
+from pydantic import BaseModel, Field, validator
+
+T = TypeVar("T")
+
+
+class JobStoreDocument(BaseModel, Generic[T]):
+    """A Pydantic model for Jobstore document."""
+
+    uuid: str = Field(
+        None, description="An unique identifier for the job. Generated automatically."
+    )
+    index: int = Field(
+        None,
+        description="The index of the job (number of times the job has been replaced).",
+    )
+    output: T = Field(
+        None,
+        description="This is a reference to the future job output.",
+    )
+    completed_at: str = Field(None, description="The time the job was completed.")
+    metadata: dict = Field(
+        None,
+        description="Metadeta information supplied by the user.",
+    )
+    hosts: List[str] = Field(
+        None,
+        description="The list of UUIDs of the hosts containing the job.",
+    )
+    name: str = Field(
+        None,
+        description="The name of the job.",
+    )
+
+    @validator("output", pre=True)
```

Comment:
Hi @hrushikesh-s, please can you update this to Pydantic2 syntax, as discussed here: https://github.com/materialsproject/jobflow/pull/425#discussion_r1332332522